src/MEDS_transforms/extract/split_and_shard_patients.py

```diff
@@ -99,31 +99,36 @@ def shard_patients[
     is_in_external_split = np.isin(patients, list(all_external_splits))
     patient_ids_to_split = patients[~is_in_external_split]
 
-    n_patients = len(patient_ids_to_split)
-
-    rng = np.random.default_rng(seed)
-    split_names_idx = rng.permutation(len(split_fracs_dict))
-    split_names = np.array(list(split_fracs_dict.keys()))[split_names_idx]
-    split_fracs = np.array([split_fracs_dict[k] for k in split_names])
-    split_lens = np.round(split_fracs[:-1] * n_patients).astype(int)
-    split_lens = np.append(split_lens, n_patients - split_lens.sum())
-
-    if split_lens.min() == 0:
-        logger.warning(
-            "Some splits are empty. Adjusting splits to ensure all splits have at least 1 patient."
-        )
-        max_split = split_lens.argmax()
-        split_lens[max_split] -= 1
-        split_lens[split_lens.argmin()] += 1
+    splits = external_splits
+
+    if n_patients := len(patient_ids_to_split):
+        rng = np.random.default_rng(seed)
+        split_names_idx = rng.permutation(len(split_fracs_dict))
+        split_names = np.array(list(split_fracs_dict.keys()))[split_names_idx]
+        split_fracs = np.array([split_fracs_dict[k] for k in split_names])
+        split_lens = np.round(split_fracs[:-1] * n_patients).astype(int)
+        split_lens = np.append(split_lens, n_patients - split_lens.sum())
+
+        if split_lens.min() == 0:
+            logger.warning(
+                "Some splits are empty. Adjusting splits to ensure all splits have at least 1 patient."
+            )
+            max_split = split_lens.argmax()
+            split_lens[max_split] -= 1
+            split_lens[split_lens.argmin()] += 1
 
-    if split_lens.min() == 0:
-        raise ValueError("Unable to adjust splits to ensure all splits have at least 1 patient.")
+        if split_lens.min() == 0:
+            raise ValueError("Unable to adjust splits to ensure all splits have at least 1 patient.")
 
-    patients = rng.permutation(patient_ids_to_split)
-    patients_per_split = np.split(patients, split_lens.cumsum())
+        patients = rng.permutation(patient_ids_to_split)
+        patients_per_split = np.split(patients, split_lens.cumsum())
 
-    splits = {k: v for k, v in zip(split_names, patients_per_split)}
-    splits = {**splits, **external_splits}
+        splits = {**{k: v for k, v in zip(split_names, patients_per_split)}, **splits}
+    else:
+        logger.info(
```

Comment:
In the event that this clause is triggered _and_ `split_fracs_dict` is not empty, we should either throw an error or, at the very least, use `logger.warning` to catch the fact that none of the splits they may have expected to create will be created. When `split_fracs_dict` is also empty, then just doing `logger.info` is fine.

Reply:
yes, you're right! it's important